tests/saitests/py3/sai_qos_tests.py

```diff
@@ -2298,7 +2300,7 @@ def runTest(self):
                                     ecn=ecn,
                                     ttl=ttl)
             dst_port_2_id = self.get_rx_port(
-                src_port_id, pkt_dst_mac, dst_port_2_ip, src_port_ip, dst_port_2_id, src_port_vlan
+                src_port_id, pkt_dst_mac2, dst_port_2_ip, src_port_ip, dst_port_2_id, src_port_vlan
```

Comment:
I understand the logic now.
It gets rx port by sending a packet and checking from which port it is received.
for t1 and t0 uplink ports, the pkt_dst_macX is router_mac for all ports.
for t0 downlink ports, it must be the ptf port's mac because it's a vlan and packets are layer 2 forwarded.